engine/servermaster/jobmanager.go

```diff
@@ -224,13 +226,25 @@ func (jm *JobManagerImpl) GetJob(ctx context.Context, req *pb.GetJobRequest) (*p
 		return nil, err
 	}
 
+	job, err := buildPBJob(masterMeta, req.IncludeConfig)
+	if err != nil {
+		return nil, err
+	}
+
 	// if job status is running, forward the request to jobmaster openapi
 	if jm.canQueryJobDetail(masterMeta) {
 		detail, err := jm.jobHTTPClient.GetJobDetail(ctx, masterMeta.Addr, req.Id)
-		setDetailToMasterMeta(masterMeta, detail, err)
+		if err != nil {
+			job.Error = &pb.Job_Error{
+				Code:    "", // TODO: extract error code from err.
+				Message: err.Error(),
+			}
+		} else {
+			job.Detail = detail
+		}
 	}
```

Comment:
Can extract this logic into a function and reuse it at L412

Reply:
Addressed in latest commits.